sdk/maps/azure-maps-search/azure/maps/search/models/_models.py

```diff
@@ -0,0 +1,448 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+# pylint: disable=unused-import,ungrouped-imports
+from typing import List, Optional
+from .._generated.models import PointOfInterest, DataSource, ReverseSearchAddressBatchItem, BatchResultSummary
+
+
+class LatLon(object):
+
+    def __init__(
+        self,
+        lat: float = None,
+        lon: float = None
+    ):
+        self._lat = lat
+        self._lon = lon
+
+    @property
+    def lat(self) -> float:
+        return self._lat
+
+    @lat.setter
+    def lat(self, value: float) -> None:
+        if not isinstance(value, float):
+            raise TypeError(
+                f'lat.setter(): got {type(value).__name__} but expected type is float')
```

Comment:
You don't need `.setter()` in the error message - this wouldn't mean much to a user.` Maybe just:
`"LatLon.lat: got ... but expected ..."`

Reply:
Is there a reason that we are doing type checking on these values at all?
This will fail if I try to set integer values - which while probably not accurate, doesn't seem like it should be rejected?
What would the server error be if this object was sent with integers rather than floats?

Reply:
> Is there a reason that we are doing type checking on these values at all? This will fail if I try to set integer values - which while probably not accurate, doesn't seem like it should be rejected? What would the server error be if this object was sent with integers rather than floats?

The API we called from TomTom would only accept `float` as `lat/lon` type, or the API will return error message for the typeError. 
But I think you made the great point to avoid validating the inputs from our SDK side as that also was stated in the Python guideline. Thanks, I will remove the type check here.